src/spec.rs

```diff
@@ -84,9 +88,19 @@ impl FromStr for Endpoints {
             .map(|addr| {
                 let span = tracing::error_span!("parse_addr", ?addr);
                 let _g = span.enter();
-                match addr.parse() {
-                    Ok(ep) => Ok(ep),
-                    Err(_) => parse_error!("invalid socket address"),
+
+                let mut parts = addr.splitn(2, '#');
```

Comment:
nit: can we document the protocol-hint syntax [here](https://github.com/linkerd/linkerd2-mock-dst/blob/a17d9f5398007d86ef9de4db05612d544b85d24e/src/main.rs#L20-L23)?